haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,134 @@
+import json
+import logging
+from abc import abstractmethod, ABC
+from typing import Optional, Dict, Union, List, Any
+
+
+from haystack.errors import AmazonBedrockConfigurationError
+from haystack.lazy_imports import LazyImport
+from haystack.nodes.prompt.invocation_layer import PromptModelInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import DefaultPromptHandler
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install farm-haystack[aws]'") as boto3_import:
+    import boto3
+    from botocore.exceptions import ClientError, BotoCoreError
+
+
+class AmazonBedrockBaseInvocationLayer(PromptModelInvocationLayer, ABC):
+    """
+    Base class for Amazon Bedrock based invocation layers.
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        max_length: Optional[str] = 2048,
+        **kwargs,
+    ):
+        super().__init__(model_name_or_path, **kwargs)
+        self.max_length = max_length
+
+        try:
+            session = AmazonBedrockBaseInvocationLayer.create_session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                aws_session_token=aws_session_token,
+                aws_region_name=aws_region_name,
+                aws_profile_name=aws_profile_name,
+            )
+            self.client = session.client("bedrock-runtime")
+        except:
+            raise AmazonBedrockConfigurationError
```

Comment:
We should make use of `raise from` here, see (https://docs.python.org/3/tutorial/errors.html#exception-chaining) to indicate that `AmazonBedrockConfigurationError` is a direct consequence of the caught error. Additionally let's add a proper error message.